base/windows/battle/BattleStatusWindow.ts

```diff
@@ -453,19 +461,54 @@ export class BattleStatusWindow{
 
     public grant_control(){
         let controls = [
-            {key: this.data.gamepad.A, on_down: this.component.trigger_state_change.bind(this.component)},
+            {key: this.data.gamepad.A, on_down: this.trigger_state_change.bind(this)},
             {key: this.data.gamepad.B, on_down: this.close.bind(this, this.close_callback)},
             {key: this.data.gamepad.L, on_down: this.previous_char.bind(this)},
             {key: this.data.gamepad.R, on_down: this.next_char.bind(this)},
-            {key: this.data.gamepad.LEFT, on_down: this.component.on_left.bind(this.component)},
-            {key: this.data.gamepad.RIGHT, on_down: this.component.on_right.bind(this.component)},
-            {key: this.data.gamepad.UP, on_down: this.component.on_up.bind(this.component)},
-            {key: this.data.gamepad.DOWN, on_down: this.component.on_down.bind(this.component)},
+            {key: this.data.gamepad.LEFT, on_down: this.current_component.on_left.bind(this.current_component)},
+            {key: this.data.gamepad.RIGHT, on_down: this.current_component.on_right.bind(this.current_component)},
+            {key: this.data.gamepad.UP, on_down: this.current_component.on_up.bind(this.current_component)},
+            {key: this.data.gamepad.DOWN, on_down: this.current_component.on_down.bind(this.current_component)},
         ];
 
         this.data.control_manager.set_control(controls, {loop_configs:{vertical: true, horizontal: true, shoulder: true}});
     }
 
+    public trigger_state_change(){
+        /*
+        this.unset_state();
+        this.current_state++;
+        this.setup_state();
+        */
+    }
+
+    private change_state(new_state:ComponentStates){
+        let pos = {line: 0, col: 0};
+        
+        if(this.current_component){
+            pos = this.current_component.current_pos;
+            this.current_component.destroy();
+            this.current_component = null;
+        }
+
+        this.current_state = new_state;
+
+        switch(this.current_state){
+            case ComponentStates.STATISTICS:
+                this.current_component = new StatusStatistics(this.game, this.data, this.window, this, pos);
```

Comment:
Would be very hard to save these components inside a variable instead of instantiating a new one every time a change it? Example: `this.current_component = this.components[this.current_state]`